csrc/ops/composite.cpp

```diff
@@ -161,7 +163,9 @@ TensorView* newForLinear(
   TensorDomain* td = IrBuilder::create<TensorDomain>(
       out_domain, TensorDomain::getContiguityFilledWith(out_domain, true));
 
-  return IrBuilder::create<TensorView>(td, input->dtype());
```

Comment:
Can we set it in the decomposition instead?

Reply:
(I assume you are referring to calling setDeviceMesh in InsertReshardingsPass.)

I'd love to understand why you prefer one way to the other. 

I think this is part of https://github.com/NVIDIA/Fuser/issues/4336, and I'm sure I haven't thought about where to setDeviceMesh as much as you have. 

I chose to setDeviceMesh in newForLinear because
1. I followed the convention of other new-a-TV methods, which could for sure be wrong in the first place. 
2. Propagating meshes is trivial and I don't mind having to do it in multiple places.

Reply:
Your thought process makes sense and the rationales are well established. I'll leave setDeviceMesh here for now just so it's consistent with the other new* methods. Feel free to revert it when working on #4336.